src/garage/tf/policies/categorical_lstm_policy2.py

```diff
@@ -0,0 +1,339 @@
+        if self._prev_actions is None or len(dones) != len(self._prev_actions):
+            self._prev_actions = np.zeros(
+                (len(dones), self.action_space.flat_dim))
+            self._prev_hiddens = np.zeros((len(dones), self._hidden_dim))
+            self._prev_cells = np.zeros((len(dones), self._hidden_dim))
+
+        self._prev_actions[dones] = 0.
+        self._prev_hiddens[dones] = self.model.networks[
+            'default'].init_hidden.eval()
+        self._prev_cells[dones] = self.model.networks[
+            'default'].init_cell.eval()
+
+    def get_action(self, observation):
+        """Return a single action.
+
+        Args:
+            observation (numpy.ndarray): Observations.
+
+        Returns:
+            int: Action given input observation.
+            dict(numpy.ndarray): Distribution parameters.
+
+        """
+        actions, agent_infos = self.get_actions([observation])
+        return actions[0], {k: v[0] for k, v in agent_infos.items()}
+
+    def get_actions(self, observations):
+        """Return multiple actions.
+
+        Args:
+            observations (numpy.ndarray): Observations.
+
+        Returns:
+            list[int]: Actions given input observations.
+            dict(numpy.ndarray): Distribution parameters.
+
+        """
+        if self._state_include_action:
+            assert self._prev_actions is not None
+            all_input = np.concatenate([observations, self._prev_actions],
+                                       axis=-1)
+        else:
+            all_input = observations
+        probs, hidden_vec, cell_vec = self._f_step_prob(
+            all_input, self._prev_hiddens, self._prev_cells)
+
+        actions = list(map(self.action_space.weighted_sample, probs))
+        prev_actions = self._prev_actions
+        self._prev_actions = self.action_space.flatten_n(actions)
+        self._prev_hiddens = hidden_vec
+        self._prev_cells = cell_vec
+        agent_info = dict(prob=probs)
+        if self._state_include_action:
+            agent_info['prev_action'] = np.copy(prev_actions)
+        return actions, agent_info
+
+    @property
+    def recurrent(self):
+        """Recurrent or not.
+
+        Returns:
+            bool: Whether policy is recurrent or not.
+
+        """
+        return True
+
+    @property
+    def distribution(self):
+        """Policy distribution.
+
+        Returns:
+            tfp.Distribution.OneHotCategorical: Policy distribution.
+
+        """
+        return self.model.networks['default'].dist
+
+    @property
+    def state_info_specs(self):
+        """State info specifcation.
+
+        Returns:
+            List[str]: keys and shapes for the information related to the
+                policy's state when taking an action.
+
+        """
+        if self._state_include_action:
+            return [
+                ('prev_action', (self._action_dim, )),
+            ]
+        return []
+
+    def clone(self, name):
```

Comment:
is clone part of the policy or model interface? should it be?

Reply:
I think it's part of model interface because eventually policy will be a model as well

Reply:
okay perhaps you should add it (or at least file an issue about it)